core/src/proxy/auth.rs

```diff
@@ -42,206 +40,160 @@ impl Interceptor for AuthInterceptor {
-        let mut num_refresh_access_token = 0;
-
-        while !exit.load(Ordering::Relaxed) {
-            let access_token_expiry: i64 = access_token
-                .lock()
-                .unwrap()
-                .expires_at_utc
-                .as_ref()
-                .map(|ts| ts.seconds)
-                .unwrap_or_default();
-            let refresh_token_expiry = refresh_token
-                .expires_at_utc
-                .as_ref()
-                .map(|ts| ts.seconds)
-                .unwrap_or_default();
-
-            let now = Utc::now().timestamp();
-
-            let should_refresh_access = access_token_expiry.checked_sub(now).ok_or_else(|| {
-                ProxyError::InvalidData("Received invalid access_token expiration".to_string())
-            })? <= REFRESH_WITHIN_SECS;
-            let should_generate_new_tokens =
-                refresh_token_expiry.checked_sub(now).ok_or_else(|| {
-                    ProxyError::InvalidData("Received invalid refresh_token expiration".to_string())
-                })? <= REFRESH_WITHIN_SECS;
-
-            match (should_refresh_access, should_generate_new_tokens) {
-                // Generate new tokens if the refresh_token is close to being expired.
-                (_, true) => {
-                    let kp = cluster_info.keypair().clone();
-
-                    let (new_access_token, new_refresh_token) =
-                        generate_auth_tokens(&mut auth_service_client, kp.as_ref()).await?;
-
-                    *access_token.lock().unwrap() = new_access_token.clone();
-                    refresh_token = new_refresh_token;
-
-                    num_full_refreshes += 1;
-                    datapoint_info!(
-                        "auth_tokens_update_loop-tokens_generated",
-                        ("url", url, String),
-                        ("count", num_full_refreshes, i64),
-                    );
-                }
-                // Invoke the refresh_access_token method if the access_token is close to being expired.
-                (true, _) => {
-                    let new_access_token =
-                        refresh_access_token(&mut auth_service_client, refresh_token.clone())
-                            .await?;
-                    *access_token.lock().unwrap() = new_access_token;
-
-                    num_refresh_access_token += 1;
-                    datapoint_info!(
-                        "auth_tokens_update_loop-refresh_access_token",
-                        ("url", url, String),
-                        ("count", num_refresh_access_token, i64),
-                    );
-                }
-                // Sleep and do nothing if neither token is close to expired,
-                (false, false) => sleep(sleep_interval).await,
-            }
-        }
-
-        Ok(())
-    }
+/// Generates an auth challenge then generates and returns validated auth tokens.
+pub async fn generate_auth_tokens(
```

Comment:
did this code change?

Reply:
just moved out of module, changed error type, and added debug statements